src/main/java/org/cprover/CProver.java

```diff
@@ -322,4 +322,17 @@ public static String classIdentifier(Object object)
   {
     return object.getClass().getCanonicalName();
   }
+
+  /**
+   * This method converts a double to a float and adds assumes that the conversion
+   * did not result any loss of precision. Calling this method is useful when we
+   * need to call a certain complicated operation on double, but we only have it
+   * implemented for floats.
+   */
+  public static float doubleToFloat(double d)
+  {
+    float converted = nondetFloat();
+    CProver.assume(d < Float.MAX_VALUE && d > -Float.MAX_VALUE && d == (double) converted);
```

Comment:
This makes sense for test generation but not really for verification. Could we change this assume for an assert?

Reply:
Hmmm. I am not sure what you suggest is better than the current approach, but I see where you're coming from, at least the user will explicitly see the violated assertion in the output. I've changed this.